docs/architecture/exception_handling.md

```diff
@@ -123,6 +123,3 @@ except mx.base.MXNetError as ex:
 d.asnumpy()
 ```
 
-### Limitation
```

Comment:
shall we call out the potential performance impact of using waitall?

Reply:
yes good point, i can add that.